commcare_connect/opportunity/models.py

```diff
@@ -370,11 +425,41 @@ def images(self):
 
 class OpportunityClaim(models.Model):
     opportunity_access = models.OneToOneField(OpportunityAccess, on_delete=models.CASCADE)
-    max_payments = models.IntegerField()
+    # to be removed
+    max_payments = models.IntegerField(null=True)
     end_date = models.DateField()
     date_claimed = models.DateField(auto_now_add=True)
 
 
+class OpportunityClaimLimit(models.Model):
+    opportunity_claim = models.ForeignKey(OpportunityClaim, on_delete=models.CASCADE)
+    payment_unit = models.ForeignKey(PaymentUnit, on_delete=models.CASCADE)
+    max_visits = models.IntegerField()
+
+    @classmethod
+    def create_claim_limits(cls, opportunity: Opportunity, claim: OpportunityClaim):
+        claim_limits_by_payment_unit = defaultdict(list)
+        claim_limits = OpportunityClaimLimit.objects.filter(
+            opportunity_claim__opportunity_access__opportunity=opportunity
+        )
+        for claim_limit in claim_limits:
+            claim_limits_by_payment_unit[claim_limit.payment_unit].append(claim_limit)
```

Comment:
related to the above question but when does this need to be a list?

Reply:
This is because we consider all users' claimlimits for an opportunity, so claimlimit for one paymentunit but for multiple users.